lib/rubocop/cop/betterment/unscoped_find.rb

```diff
@@ -55,10 +56,12 @@ def on_send(node)
         private
 
         def graphql_namespace?(node)
+          return true if processed_source.path&.match(GRAPHQL_PATTERN)
+
           node
             .ancestors
             .select { |n| n.class_type? || n.module_type? }
-            .any? { |n| n.identifier.to_s.match(/\bGraphQL\b/i) }
+            .any? { |n| n.identifier.to_s.match(GRAPHQL_PATTERN) }
```

Comment:
__suggestion:__ I would suggest using `String#match?` because it satisfies the requirements, but also has a simpler return value; and is a "simpler primitive".

In code I tend to try to follow [the principle of least powert](https://blog.codinghorror.com/the-principle-of-least-power/), which means that if I have 2 choices for a language construct, and both satisfy the requirements, I will choose the approach that has "less degrees of freedom" -- that is, it either has fewer or more constrained inputs or  a more constrained return type.

In this case we are moving from a method with the MatchData return value to a Boolean. Sure, the evaluation of MatchData works due to how ruby evaluates non-nil objects in boolean context, but the Boolean return value of `#match?` has even fewer ways it can be used -- in a future refactoring the developer can see our intentions more clearly here is a true/false match, and not something more where we evaluate MatchData.